### DISCUSSION TITLE
socketIO async_mode=eventlet unable to display some logs

### DISCUSSION THREAD

**Discussion by liuqijie6** [miguelgrinberg/Flask-SocketIO#1930] [Q&A] [answered]
socketIO config
```
socket_io = SocketIO(cors_allowed_origins='*',
                     async_mode="threading",  # eventlet  gevent # threading
                     # engineio_logger=True,
                     # logger=True,
                     message_queue=f'redis://:{password}@{host}:{port}/0',
                     )
```
**app.py**
```
__all__ = ["app"]
import eventlet
eventlet.monkey_patch()
from __init__ import create_app
from units.extensions import socket_io
from config import CONFIG

app = create_app()

if __name__ == "__main__":
    host, port = CONFIG.get("web_server", "host"), CONFIG.get("web_server", "port")
    # print(app.url_map)
    # app.run(host=host, port=port, threaded=False, debug=False)
    socket_io.run(app, host=host, port=port, debug=False)

```
if async_mode = eventlet , cant print this.
if async_mode = threading , can print this.
![image](https://user-images.githubusercontent.com/52588515/211711669-49d79fe2-08ad-4ac4-89b2-d0b83007698c.png)

**Comment by miguelgrinberg** [comment, marked as answer]
I think this is a misunderstanding on your part. When you use `threading`, you are using the Flask development web server. If you change the async mode to `eventlet` you are using eventlet's web server. The choice to log or not to log depends on each web server. The eventlet server uses the `log_output` option in the `socket_io.run()` function to determine if logging is enabled or not.
